modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchHandler.kt

```diff
@@ -120,28 +120,40 @@ class SearchHandler @Inject constructor(
             } else {
                 analyticsTracker.track(AnalyticsEvent.SEARCH_PERFORMED, AnalyticsProp.sourceMap(source))
                 loadingObservable.accept(true)
+
+                var globalSearch = GlobalServerSearch(searchTerm = it)
                 val podcastServerSearch = serverManager
                     .searchForPodcastsRx(it)
+                    .map { podcastSearch ->
+                        globalSearch = globalSearch.copy(podcastSearch = podcastSearch)
+                        globalSearch
+                    }
                     .subscribeOn(Schedulers.io())
                     .toObservable()
 
-                val episodesServerSearch = cacheServerManager
-                    .searchEpisodes(it)
+                if (BuildConfig.SEARCH_IMPROVEMENTS_ENABLED) {
+                    val episodesServerSearch = cacheServerManager
+                        .searchEpisodes(it)
+                        .map { episodeSearch ->
+                            globalSearch = globalSearch.copy(episodeSearch = episodeSearch)
+                            globalSearch
+                        }
+                        .subscribeOn(Schedulers.io())
+                        .toObservable()
+
+                    podcastServerSearch.mergeWith(episodesServerSearch)
+                } else {
+                    podcastServerSearch
+                }
                     .subscribeOn(Schedulers.io())
```

Comment:
With this `.subscribeOn(Schedulers.io()` do we also need the ones on lines 131 and 141? I'm wondering if this might be the only one we need.

Reply:
I verified that those were not needed so removed them in 13c6d2f.